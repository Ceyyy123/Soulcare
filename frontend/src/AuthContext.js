import React, { createContext, useContext, useState, useEffect } from 'react';

const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [isAuthenticated, setIsAuthenticated] = useState(false);

  // Überprüfe bei der Initialisierung, ob ein Token vorhanden ist
  useEffect(() => {
    const token = localStorage.getItem('token');
    if (token) {
      setIsAuthenticated(true);
    }
  }, []);

  const login = (token) => {
    // Speichere den Token im localStorage
    localStorage.setItem('token', token);
    setIsAuthenticated(true);
  };

  const logout = () => {
    // Entferne den Token aus dem localStorage
    localStorage.removeItem('token');
    setIsAuthenticated(false);
  };

  return (
    <AuthContext.Provider value={{ isAuthenticated, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
};
export const signupUser = async (email, password) => {
  const response = await fetch('https://soulcare-mu.vercel.app/api/users/signup', { 
      method: 'POST',
      headers: {
          'Content-Type': 'application/json',
      },
      body: JSON.stringify({ email, password }),
  });

  if (!response.ok) {
      throw new Error('Failed to signup user');
  }

  return true;
};


export const useAuth = () => useContext(AuthContext);
