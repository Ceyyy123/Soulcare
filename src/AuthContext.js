import React, { createContext, useContext, useState, useEffect } from 'react';

const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [isAuthenticated, setIsAuthenticated] = useState(false);

  // Überprüfe bei der Initialisierung, ob ein Token vorhanden ist
  useEffect(() => {
    const token = localStorage.getItem('token');
    if (token) {
      setIsAuthenticated(true);
    }
  }, []);

  const login = (token) => {
    // Speichere den Token im localStorage
    localStorage.setItem('token', token);
    setIsAuthenticated(true);
  };

  const logout = () => {
    // Entferne den Token aus dem localStorage
    localStorage.removeItem('token');
    setIsAuthenticated(false);
  };

  return (
    <AuthContext.Provider value={{ isAuthenticated, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
};


export const useAuth = () => useContext(AuthContext);
