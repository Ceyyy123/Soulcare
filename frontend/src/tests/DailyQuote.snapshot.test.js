import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import DailyQuote from '../pages/dailyquote';
import { AuthProvider } from '../AuthContext';

jest.mock('next/link', () => ({ children, ...props }) => (
  <a {...props}>{children}</a>
));

test('renders DailyQuote correctly', () => {
  const { asFragment } = render(
    <AuthProvider>
      <DailyQuote />
    </AuthProvider>
  );

  // Erstellt oder überprüft den Snapshot
  expect(asFragment()).toMatchSnapshot();
});
