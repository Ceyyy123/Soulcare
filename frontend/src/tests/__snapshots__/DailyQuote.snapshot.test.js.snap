// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DailyQuote correctly 1`] = `
<DocumentFragment>
  <div
    class="quoteContainer"
  >
    <p>
      Bitte melde dich an, um das inspirierende Zitat des Tages und die Wohlf√ºhl-Tipps zu sehen.
    </p>
    <a
      class="resourceButton"
      href="/login"
    >
      Zum Login
    </a>
  </div>
</DocumentFragment>
`;
